### ISSUE TITLE
Output files & filtering

### ISSUE THREAD

**Issue by ATpoint** [rhysnewell/ChIP-R#1]
Hello,

thanks for putting this tool together.

Could you please elaborate on what exactly the output files are and how one would need to filter them?
In contrast to the README I get two files, `*_all.bed` and `*_optimal.bed` (but not T1/T2).
The command was simply `chipr -i rep1.narrowPeak rep2.narrowPeak -o out` for a normal transcription factor ChIP-seq.
What is the "optimal" file here compared to "all"?
I am unsure how to obtain the final list of reproducible peaks (and from which file).
Do I filter any of these files based on FDR ($9)? 
Since both files contain entries with FDR > 0.05 (with --alpha left at default 0.05), what is the relationship between FDR and alpha (if there is any), and when would it make sense to change alpha?

Hope you can clarify, thank you for your time!

**Comment by j-andrews7** [comment]
I would also like to know the answer to this.

**Comment by nchambwe** [comment]
Yes - me too!  Had a conference with @j-andrews7 about this earlier today.

**Comment by millerh1** [comment]
I also had this question. The end of `rankprod.py` [here](https://github.com/rhysnewell/ChIP-R/blob/32ad6f9694e279624a044b0e3793c5da1182eeba/chipr/rankprod.py#L1707) seems to indicate that primary is t1 and secondary is anything which didn't meet the threshold for t1. I think T1 just means the rank product bound pval is less than whatever the nbinom alpha was determined to be.

**Comment by ZunpengLiu** [comment]
Got the same question. Looking forward to the answer here.

**Comment by chiefcat** [comment]
From the published paper supplementary figure 1 legend:

After all test fragments have been filtered and/or collapsed into output peaks, two output files are produced: “optimal” for peaks p ≤ θ (where θ is the threshold suggested by the binomial test) and “all” containing all peaks regardless of p.

https://ars.els-cdn.com/content/image/1-s2.0-S0888754321001531-mmc1.pdf

**Comment by yeroslaviz** [comment]
is there any progress here with an answer?

thx

**Comment by BioLaoXu** [comment]
I had the same issue, by looking at the source code, I found that the difference between `primary` and `secondary` is whether it is less than the significance threshold (`--alpha`, default: 0.05, similar to idr tool), the pValue of the `primary` peak is less than or equal to alpha, and the pValue of the `secondary `peak is greater than the alpha; In addition, by looking at the file contents of `*_all.bed` and `*_optimal.bed`, it is found that `*_optimal.bed` is a true subset of `*_all.bed`.

By the way, my personal understanding, in order to calculate a consistency score like IDR
```
#method1
paste <(wc -l ./*_optimal.bed|cut -f1 -d ' ') <(wc -l *_all.bed|cut -f1 -d ' ') | awk '{ print $1/$2*100 }'
#method2
cat *_log.txt|head -3|sed -r 's/.* //'|sed -r 's/.*://'|xargs -n3|awk '{ print $1/$3*100 }'
#method
fetch from teminal log information directly
```
